src/dune/foreign.mli

```diff
@@ -70,6 +68,59 @@ val possible_sources :
   -> dune_version:Dune_lang.Syntax.Version.t
   -> string list
 
+(** Foreign archives appear in the [(foreign_archives ...)] field of libraries
+    and executables, for example [(foreign_archives some/dir/lib)]. When parsing
+    such fields, we separate the directory [some/dir] from the name [lib] of the
+    archive and store them as a [{dir : Dir.t; name : Name.t}] record. *)
+module Archive : sig
+  (** Archive names appear as part of the [(foreign_archives ...)] fields, as
+      well as in the [(archive_name ...)] field of the [(foreign_library ...)]
+      stanza. For example, both [(foreign_archives some/dir/lib)] and
+      [(archive_name lib)] specify the same archive name [lib]. Archive names
+      are not allowed to contain path separators. *)
+  module Name : sig
+    type t
+
+    module Map : Map.S with type key = t
+
+    val to_dyn : t -> Dyn.t
+
+    val to_string : t -> string
+
+    val path : dir:Path.Build.t -> t -> Path.Build.t
+
+    val decode : t Dune_lang.Decoder.t
+
+    val stubs : string -> t
+
+    val lib_file : t -> dir:Path.Build.t -> ext_lib:string -> Path.Build.t
+
+    val dll_file : t -> dir:Path.Build.t -> ext_dll:string -> Path.Build.t
+  end
+
+  (** Archive directories can appear as part of the [(foreign_archives ...)]
+      fields. For example, in [(foreign_archives some/dir/lib1 lib2)], the
+      archive [some/dir/lib1] has the directory [some/dir], whereas the archive
+      [lib2] does not specify the directory and is assumed to be located in [.]. *)
+  module Dir : sig
```

Comment:
Does this module need to be in the mli at this point?

Reply:
Indeed, I've dropped it.